helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -92,19 +99,51 @@ public ZkBaseDataAccessor(HelixZkClient zkClient) {
     _zkClient = zkClient;
   }
 
+  /**
+   * Lazy initialization is used to reduce the cost of unnecessary opened zkConnections
+   * @return singleton instance of non-ZnRecord ZkClient
+   */
+  private HelixZkClient getNonZNRecordZkClient() {
+    if (_nonZNRecordZkClient == null) {
+      synchronized (ZkBaseDataAccessor.class) {
```

Comment:
Is there a reason we're synchronizing on this? I think when we synchronize on the class, we usually do `synchronized (this)`?

My previous comment is suggesting that we synchronize on the nonZNRecordZkClient object.

Reply:
synchronized on the nullable object could lead to NPE, I like the synchronized(this) idea. Sounds good!

Reply:
Great. You're correct - let's use synchronized(this) since the ZkClient could be null in this case.